packages/remix-react/components.tsx

```diff
@@ -905,27 +938,7 @@ import(${JSON.stringify(manifest.entry.module)});`;
     }
   }
 
-  // avoid waterfall when importing the next route module
-  let nextMatches = React.useMemo(() => {
-    if (navigation.location) {
-      // FIXME: can probably use transitionManager `nextMatches`
-      let matches = matchRoutes(
-        router.routes,
-        navigation.location,
-        router.basename
-      );
-      invariant(
-        matches,
-        `No routes match path "${navigation.location.pathname}"`
-      );
-      return matches;
-    }
-
-    return [];
-  }, [navigation.location, router.routes, router.basename]);
-
```

Comment:
@jacob-ebey This seems like dead code to me?  It only returns a non-empty array when we're in the middle of a client-side `navigation` (which implies we've already hydrated), but the `routePreloads` it gets concatenated into below is only ever rendered when we're not hydrated.  I think it's likely leftover from the `TransitionManager` implementation and maybe before `<PrefetchPageLinks>`?